### ISSUE TITLE
Crash when placing tiles in tile layer with no Default tileset assigned

### ISSUE THREAD

**Issue by PeterVerzijl** [deepnight/ldtk#1047]
Crash when placing any tile in a tile layer which has its Default tileset set to nothing (--Tileset required--).

(I don't know why this was set to nothing, it had a tileset set before and the layer was still rendering correctly)

Stack:
```
LDtk version: 1.5.2-64bits
Cannot read properties of null (reading 'averageColorsCache')
TypeError
TypeError: Cannot read properties of null (reading 'averageColorsCache')
    at data_inst_LayerInstance.addGridTile (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:53729:21)
    at tool_lt_TileTool.drawSelectionAt (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:272268:9)
    at tool_lt_TileTool.useAtInterpolatedGrid (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:270683:13)
    at tool_lt_TileTool.useAt (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:21544:15)
    at tool_lt_TileTool.startUsing (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:18555:57)
    at tool_lt_TileTool.startUsing (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:270624:39)
    at page_Editor.onMouseDown (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:238169:25)
    at page_Editor.onHeapsEvent (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:238080:9)
    at h2d_Scene.dispatchListeners (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:142756:4)
    at hxd_SceneEvents.dispatchListeners (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:189933:6)
```
Processes:
```
#0 App
 |--- #3 page.Editor
       |--- #4 misc.FileWatcher
       |--- #5 display.WorldRender
       |--- #6 display.LevelRender
       |--- #7 display.Camera
       |--- #8 display.Rulers[--]
       |--- #9 tool.SelectionTool[--]
       |--- #10 tool.lt.DoNothing[--]
       |--- #11 WorldTool
       |--- #12 tool.PanView[--]
       |--- #13 tool.lt.TileTool [PAUSED][--]
       |--- #38 display.EntityRender
       |--- #39 display.EntityRender
       |--- #40 display.EntityRender
       |--- #41 display.EntityRender
       |--- #42 display.EntityRender
       |--- #43 display.EntityRender
       |--- #44 display.EntityRender
       |--- #45 display.EntityRender
       |--- #46 display.EntityRender
       |--- #47 display.EntityRender
       |--- #48 display.EntityRender
       |--- #50 tool.lt.TileTool[Add] [RUNNING]
```

Log:
```
[RENDER]       Rendering world level data.Level#499a6100-8990-11ee-b686-8bde8389dfc0 "Level_5"...
[RENDER]       Rendering world bg...
[USERACTION]   Opened panel #37 ui.modal.panel.WorldPanel
[FILE]         Loading HTML template worldPanel
[FILE]         Loading HTML template levelInstanceForm
[RENDER]       Rendering world level data.Level#7b785082-8990-11ee-bdac-cf371c62941c "Level_0"...
[RENDER]       Rendering world level fields data.Level#7b785082-8990-11ee-bdac-cf371c62941c "Level_0"...
[RENDER]       Rendering world level fields data.Level#d8b9b680-8990-11ee-bdac-93cef95db54e "Level_1"...
[RENDER]       Rendering world level fields data.Level#499a6100-8990-11ee-b686-8bde8389dfc0 "Level_5"...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world level data.Level#d8b9b680-8990-11ee-bdac-93cef95db54e "Level_1"...
[RENDER]       Rendering world level fields data.Level#0a564130-8990-11ee-bdac-b5288532565d "Level_2"...
[RENDER]       Rendering world level data.Level#0a564130-8990-11ee-bdac-b5288532565d "Level_2"...
[RENDER]       Rendering world level data.Level#499a6100-8990-11ee-b686-8bde8389dfc0 "Level_5"...
[RENDER]       Rendering world level data.Level#40104390-8990-11ee-bdac-9df0d2f1f4fc "Level_3"...
[RENDER]       Rendering world level fields data.Level#40104390-8990-11ee-bdac-9df0d2f1f4fc "Level_3"...
[RENDER]       Rendering world level fields data.Level#67f52a50-8990-11ee-bdac-ebd83c2cfed7 "Level_4"...
[RENDER]       Rendering world level fields data.Level#77714da0-8990-11ee-b686-813fb2e7c7b7 "Level_6"...
[RENDER]       Rendering world level data.Level#67f52a50-8990-11ee-bdac-ebd83c2cfed7 "Level_4"...
[RENDER]       Rendering world level fields data.Level#9d9232b0-8990-11ee-b686-598d72d286c5 "Level_7"...
[RENDER]       Rendering world level fields data.Level#378cdeb0-8990-11ee-84c5-1d4bc3bb77b8 "Level_8"...
[RENDER]       Rendering world level data.Level#77714da0-8990-11ee-b686-813fb2e7c7b7 "Level_6"...
[RENDER]       Rendering world level fields data.Level#ce59ed40-8990-11ee-84c5-8b1a5680b8a9 "Level_9"...
[RENDER]       Rendering world level data.Level#9d9232b0-8990-11ee-b686-598d72d286c5 "Level_7"...
[RENDER]       Rendering world level data.Level#378cdeb0-8990-11ee-84c5-1d4bc3bb77b8 "Level_8"...
[RENDER]       Rendering world level data.Level#ce59ed40-8990-11ee-84c5-8b1a5680b8a9 "Level_9"...
[EVENT]        LevelSelected 58
[USERACTION]   Opened level data.Level#378cdeb0-8990-11ee-84c5-1d4bc3bb77b8 "Level_8"
[TIMELINE]     Garbage collected level: #0
[USERACTION]   Closed panel #37 ui.modal.panel.WorldPanel
[WARNING]      Full level render requested
[RENDER]       Rendering world bg...
[RENDER]       Rendering world level data.Level#67f52a50-8990-11ee-bdac-ebd83c2cfed7 "Level_4"...
[RENDER]       Rendering world level fields data.Level#378cdeb0-8990-11ee-84c5-1d4bc3bb77b8 "Level_8"...
[RENDER]       Rendering world level data.Level#378cdeb0-8990-11ee-84c5-1d4bc3bb77b8 "Level_8"...
[RENDER]       Rendering world level data.Level#ce59ed40-8990-11ee-84c5-8b1a5680b8a9 "Level_9"...
[RENDER]       Rendering world bg...
[EVENT]        LayerInstanceSelected 65
[RENDER]       Rendered level UI
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[GENERAL]      Page started: page.CrashReport()
[FILE]         Loading page template: crashReport from C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/tpl/pages/crashReport.html
[ERROR]        Cannot read properties of null (reading 'averageColorsCache') (TypeError)
[ERROR]        TypeError: Cannot read properties of null (reading 'averageColorsCache')
    at data_inst_LayerInstance.addGridTile (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:53729:21)
    at tool_lt_TileTool.drawSelectionAt (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:272268:9)
    at tool_lt_TileTool.useAtInterpolatedGrid (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:270683:13)
    at tool_lt_TileTool.useAt (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:21544:15)
    at tool_lt_TileTool.startUsing (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:18555:57)
    at tool_lt_TileTool.startUsing (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:270624:39)
    at page_Editor.onMouseDown (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:238169:25)
    at page_Editor.onHeapsEvent (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:238080:9)
    at h2d_Scene.dispatchListeners (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:142756:4)
    at hxd_SceneEvents.dispatchListeners (file:///C:/Users/pverzijl/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:189933:6)

[GENERAL]      
#0 App
 |--- #3 page.Editor
       |--- #4 misc.FileWatcher
       |--- #5 display.WorldRender
       |--- #6 display.LevelRender
       |--- #7 display.Camera
       |--- #8 display.Rulers[--]
       |--- #9 tool.SelectionTool[--]
       |--- #10 tool.lt.DoNothing[--]
       |--- #11 WorldTool
       |--- #12 tool.PanView[--]
       |--- #13 tool.lt.TileTool [PAUSED][--]
       |--- #38 display.EntityRender
       |--- #39 display.EntityRender
       |--- #40 display.EntityRender
       |--- #41 display.EntityRender
       |--- #42 display.EntityRender
       |--- #43 display.EntityRender
       |--- #44 display.EntityRender
       |--- #45 display.EntityRender
       |--- #46 display.EntityRender
       |--- #47 display.EntityRender
       |--- #48 display.EntityRender
       |--- #50 tool.lt.TileTool[Add] [RUNNING]
 |--- #53 page.CrashReport

[FILE]         Cleared all file watches
```

**Comment by deepnight** [comment]
Thanks for reporting!

Note for myself: crash happens in layers without default Tileset, but a custom tileset selected in the Main Panel on the left side.

**Comment by deepnight** [comment]
Fixed in 1.5.3
